### PR TITLE
support OPT_X_TLS_PEERCERT

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: Modules/options.c [hunk 1]
```diff
@@ -399,7 +401,20 @@ LDAP_get_option(LDAPObject *self, int option)
         v = LDAPControls_to_List(lcs);
         ldap_controls_free(lcs);
         return v;
-
+#ifdef LDAP_OPT_X_TLS_PEERCERT
+    case LDAP_OPT_X_TLS_PEERCERT:
```
Comment (graingert) [inline Modules/options.c:405]: ah this is only in openldap >=2.5.0

**Comment by graingert** [comment]
Is there a way to get CI to test against the openldap alpha releases?

**Comment by encukou** [comment]
> Is there a way to get CI to test against the openldap alpha releases?

Not really, but you add a test that's skipped based on the libldap version, a manual test should be enough.

**Comment by graingert** [comment]
I don't know how to set up the openldap alpha either

**Comment by quanah** [comment]
OpenLDAP 2.5 is the official current release, 2.4 is EOL, python-ldap should likely figure out how to support the current release.

**Comment by encukou** [comment]
Is there anything in python-ldap that doesn't work with OpenLDAP 2.5?

Which distros ship it? The major ones, including Arch, seem to be on 2.4 – that doesn't look EOL to me.

**Comment by quanah** [comment]
> 
> 
> Is there anything in python-ldap that doesn't work with OpenLDAP 2.5?
> 
> Which distros ship it? The major ones, including Arch, seem to be on 2.4 – that doesn't look EOL to me.

From the OpenLDAP project perspective, it is end of life.  What distributions do is not something the OpenLDAP project tracks.

**Comment by encukou** [comment]
> From the OpenLDAP project perspective, it is end of life. 

Do you have any source for that info? The [2.4.59 release announcement](https://lists.openldap.org/hyperkitty/list/openldap-announce@openldap.org/thread/UREG3X43QNEQGAVSKLHKO752QVVNRQPP/) from a few months ago doesn't mention the 2.4.x line being EOL.

**Comment by quanah** [comment]
> 
> 
> > From the OpenLDAP project perspective, it is end of life.
> 
> Do you have any source for that info? 

Well, given I'm the release manager for the OpenLDAP project, I'd say that's a pretty decent source of info. We have no plans on any further 2.4 releases unless a major CVE crops up.  Our current focus in on getting OpenLDAP 2.6 out in the next quarter.

**Comment by graingert** [comment]
Closing in favour of #427

**Pull request closed without merging**